adding sphinx gallery

choldgraf opened the pull request:
OK this one generates a full sphinx-gallery deployment with 2 examples and a more tutorial-style example for sphinx-gallery. LMKWYT.

choldgraf committed:
adding sphinx gallery

choldgraf commented:
@arokem

arokem commented:
Looks great! 

Travis seems to be angry because of pep8 issues in the examples. Maybe we can add that to things that get ignored here: https://github.com/uwescience/shablona/blob/master/.travis.yml#L54

choldgraf committed:
makefile and flake fixing

choldgraf commented:
Are you friendly to adding a `Makefile` with some flake commands in it to the root? I ran this and made sure it didn't error with anything. See latest commit.

choldgraf commented:
looks like flake is failing...it's passing on my computer so it must be a difference in rules. what would you like me to do?

arokem commented on examples/model_fitting/plot_function_stability.py:
```diff
@@ -0,0 +1,72 @@
+fig, ax = plt.subplots()
+for n_level, i_coefs, percs in zip(noise_levels, coefs, percentiles):
+    ax.scatter(np.repeat(n_level, len(i_coefs)), i_coefs)
+    ax.hlines(percs, n_level-.2, n_level+.2, lw=2, color='r', alpha=.6)
```
I believe `flake8` wants whitespace around the arithmetic operators here: 

    ax.hlines(percs, n_level - .2, n_level + .2, lw=2, color='r', alpha=.6)

instead of: 

    ax.hlines(percs, n_level-.2, n_level+.2, lw=2, color='r', alpha=.6)

choldgraf replied:
ah yeah, that was dumb - will fix this

arokem commented:
I am not sure what the Makefile is supposed to do. Could you please explain? Either way, it has an `mne` in it, which should probably be changed...

choldgraf commented:
the makefile is just to be able to easily run flake, so I just call `make flake` and it does the pyflakes check. Maybe there's an easier way to do this though? As you noted, this is just how MNE does it. I thought it'd be useful to have an example of how a person would do flake checks.

arokem commented:
Gotcha. I usually copy this line from the travis configuration file into the terminal: 

https://github.com/uwescience/shablona/blob/master/.travis.yml#L54

:-)

choldgraf committed:
updating flake function

choldgraf commented:
Cool - I've added that flake line to the makefile and fixed the flake problems. Agree that it's easy to just copy/paste the flake line, but I think it's nice to have it in a makefile because it lowers the barrier to actually doing it :) that way my workflow is usually make changes -> run `make flake` -> commit / push

arokem commented:
OK - this looks good to me. Merging. 

Thanks!
